Recipes/Recipes.API/Controllers/RecipesController.cs

```diff
@@ -0,0 +1,18 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Recipes.DataAccess.Interfaces;
+using Recipes.Public;
+
+namespace Recipes.API.Controllers;
+
+[ApiController]
+[Route("recipes")]
+public class RecipesController : ControllerBase
+{
+    private readonly IGenericRepository<Recipe> _recipesRepository;
+
+    public RecipesController(IGenericRepository<Recipe> recipesRepository)
+    {
+        _recipesRepository = recipesRepository;
+    }
+
```

Comment:
Could we move the `/recipes` endpoint to the controller from Program.cs with this PR already? We would make sure everything works as expected.

Reply:
Sure, going to do that right now

Reply:
I implemented the generic repository functions and added a single endpoint to get all recipes in the controller. However, I seem to be getting an exception when trying to retrieve the recipes. The exception happens even before it enters the endpoint:

![paveikslas](https://github.com/BenasB/vu-psk/assets/90962230/a9f87b94-c274-4f86-a2e2-5f0aee09395f)

I re-ran the `docker-compose` file and tried to update the database. I got this error:

![paveikslas](https://github.com/BenasB/vu-psk/assets/90962230/1c45e5a1-3ab2-41a2-9cb6-ccb29d0cfab7)

Reply:
DbSet has previously been injected to the repository, which couldn't be resolved. I pushed a commit that fixes it.